### PR TITLE
[2206] Iterate UI for new cycle

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
New recruitment cycle

### Changes proposed in this pull request
- Add next recruitment_cycle
- Refactor org#show page

### Guidance to review
🚢 

# After
localhost_3000_organisations_2AT(Laptop with MDPI screen) (1)

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Interate publish frontend for upcoming cycle](https://trello.com/c/ek738gkT/2206-interate-publish-frontend-for-upcoming-cycle)

File: app/views/providers/show.html.erb [hunk 1]
```diff
@@ -7,48 +7,44 @@
...
+    </h3>
+    <p class="govuk-body">View your courses and locations from the 2019 – 2020 recruitment cycle.</p>
 
   </div>
   <aside class="govuk-grid-column-one-third">
     <div class="app-related">
-      <h2 class="govuk-heading-m">New features in Publish</h2>
-      <p class="govuk-body"><%= govuk_link_to "View a list of new features", new_features_path %> we plan to introduce in the coming months.</p>
+
+      <h2 class="govuk-heading-m govuk-!-margin-bottom-2"><%= govuk_link_to "Request access", request_access_provider_path(code: @provider.provider_code) %></h2>
```
Comment (fofr) [inline app/views/providers/show.html.erb]: These headings should now be `govuk-heading-s`

File: app/views/providers/show.html.erb [hunk 2]
```diff
@@ -7,48 +7,44 @@
...
+    <p class="govuk-body govuk-!-margin-bottom-8">Candidates can only apply to published courses.</p>
+
+    <%= render partial: 'recruitment_cycles/about_organisation', locals: { provider: @provider, year: '2020' } %>
+    <%= render partial: 'recruitment_cycles/courses_info', locals: { provider: @provider, year: '2020' } %>
+    <%= render partial: 'recruitment_cycles/locations_info', locals: { provider: @provider, year: '2020' } %>
+
+    <h2 class="govuk-heading-l">Previous cycles</h2>
+
+    <h3 class="govuk-heading-m">
+      <%= link_to '2019 - 2020',
```
Comment (fofr) [inline app/views/providers/show.html.erb]: This link doesn’t work. It keeps redirecting back to the organisation page.

File: app/views/providers/show.html.erb (hunk 1 above)
Comment (dankmitchell) [reply app/views/providers/show.html.erb]: Done

File: app/views/providers/show.html.erb (hunk 2 above)
Comment (dankmitchell) [reply app/views/providers/show.html.erb]: Fixed this bug

File: app/views/recruitment_cycles/show.html.erb [hunk 3]
```diff
@@ -7,19 +7,13 @@
...
-        Prepare your courses and locations for the next cycle. All your courses, locations and details have been copied from the current&nbsp;cycle.
-      </p>
-    <% else %>
-      <p class="govuk-body-l">
-        Manage courses, locations and vacancies in the current&nbsp;cycle.
-      </p>
-    <% end %>
+    <p class="govuk-body-l">
+      View your courses and locations from the <%= @recruitment_cycle.year_range %> recruitment cycle.
+    </p>
```
Comment (fofr) [inline app/views/recruitment_cycles/show.html.erb]: Can we avoid losing this text with some logic?

```
if next cycle
  existing next text
elsif current cycle
  existing current text
else
  new text
```

File: app/views/recruitment_cycles/show.html.erb [hunk 4]
```diff
@@ -7,19 +7,13 @@
...
     <hr class="govuk-section-break govuk-section-break--m">
 
-    <%= render partial: 'recruitment_cycles/about_organisation', locals: { provider: @provider, year: params[:year] } %>
-    <%= render partial: 'recruitment_cycles/courses_info', locals: { provider: @provider, year: params[:year] } %>
-    <%= render partial: 'recruitment_cycles/locations_info', locals: { provider: @provider, year: params[:year] } %>
+    <h2 class="govuk-heading-m"><%= govuk_link_to "About your organisation", details_provider_recruitment_cycle_path(@provider.provider_code, params[:year]) %></h2>
+    <h2 class="govuk-heading-m"><%= link_to "Courses", provider_recruitment_cycle_courses_path(@provider.provider_code, params[:year]), class: "govuk-link", data: { qa: 'provider__courses' } %></h2>
+    <h2 class="govuk-heading-m"><%= link_to "Locations", provider_recruitment_cycle_sites_path(@provider.provider_code, params[:year]), class: "govuk-link", data: { qa: 'provider__locations' } %></h2>
```
Comment (fofr) [inline app/views/recruitment_cycles/show.html.erb]: Same here, we should use the partials if next or current.

File: config/locales/en.yml [hunk 5]
```diff
@@ -52,9 +52,9 @@ en:
         help: "You will consider candidates with a pending GCSE. But UCAS will block applications from someone who needs to take an equivalency test."
         details_label: "Taking"
       equivalence_test:
-        label: "3: Equivalence test"
+        label: "3: Equivalency test"
```
Comment (fofr) [inline config/locales/en.yml]: Why is this change here?

File: config/routes.rb [hunk 6]
```diff
@@ -16,29 +16,29 @@
...
+    get "/courses/:course_code/preview", to: redirect("/organisations/%{provider_code}/2020/courses/%{course_code}/preview")
 
     get "/request-access", on: :member, to: "access_requests#new"
     post "/request-access", on: :member, to: "access_requests#create"
 
     get "/ucas-contacts", on: :member, to: "ucas_contacts#index"
 
     # TODO: Extract year constraint to future proof for future cycles
-    resources :recruitment_cycles, param: :year, constraints: { year: /2019|2020/ }, path: "", only: :show do
+    resources :recruitment_cycles, param: :year, constraints: { year: /2019|2020|2021/ }, path: "", only: :show do
```
Comment (fofr) [inline config/routes.rb]: This will allow users to visit the 2021 paths when nothing is there, no?

File: spec/decorators/course_decorator_spec.rb [hunk 7]
```diff
@@ -58,7 +58,7 @@
 
   context "recruitment cycles" do
     before do
-      allow(Settings).to receive(:current_cycle).and_return(2019)
+      allow(Settings).to receive(:current_cycle).and_return(2020)
```
Comment (fofr) [inline spec/decorators/course_decorator_spec.rb]: Not sure why 2019 needs to change here.

File: spec/features/courses/details_spec.rb [hunk 8]
```diff
@@ -140,7 +140,7 @@
       course_details_page.load_with_course(course)
       expect(course_details_page).to have_entry_requirements
       expect(course_details_page.entry_requirements).to have_content("Maths GCSE: Taking")
-      expect(course_details_page.entry_requirements).to have_content("Science GCSE: Equivalence test")
+      expect(course_details_page.entry_requirements).to have_content("Science GCSE: Equivalency test")
```
Comment (fofr) [inline spec/features/courses/details_spec.rb]: Why in this PR?

File: app/views/recruitment_cycles/show.html.erb (hunk 4 above)
Comment (dankmitchell) [reply app/views/recruitment_cycles/show.html.erb]: The partials have contextual information. The design was a title link only...

File: config/locales/en.yml (hunk 5 above)
Comment (dankmitchell) [reply config/locales/en.yml]: Paul G asked for it so done as a sperate commit

File: config/routes.rb (hunk 6 above)
Comment (dankmitchell) [reply config/routes.rb]: Correct but I think that's fine since users don't know about it yet. We have a card todo rollover properly so I think we should tighten this up then.

File: spec/features/courses/details_spec.rb (hunk 8 above)
Comment (dankmitchell) [reply spec/features/courses/details_spec.rb]: Paul G asked for it so done as a sperate commit

File: app/views/recruitment_cycles/show.html.erb (hunk 4 above)
Comment (fofr) [reply app/views/recruitment_cycles/show.html.erb]: But in the future, when we go into a rollover period again, we'll want to reshow that context on the cycles page.

**Comment by fofr** [comment]
I'm happy with design. I think someone else needs to review code/test changes.

File: config/settings.yml [hunk 9]
```diff
@@ -20,7 +20,8 @@ search_ui:
   # URL of the C# search ui app (search-and-compare-ui)
   base_url: https://localhost:5000
 google_forms:
-  current_cycle:
+  # TODO: Update to new Google Form until courses#new is created inside the app
```
Comment (defong) [inline config/settings.yml]: Done?

File: spec/requests/sites_spec.rb [hunk 10]
```diff
@@ -10,21 +10,21 @@
   end
 
   describe "GET index" do
-    it "redirects to /2019/locations" do
+    it "redirects to /2020/locations" do
```
Comment (defong) [inline spec/requests/sites_spec.rb]: may be just stick 2020 in a variable to be shared for the others

current_recruitment_cycle_year = 2020

File: spec/site_prism/page_objects/page/organisations/organisation_page.rb [hunk 11]
```diff
@@ -6,8 +6,8 @@ class OrganisationPage < PageObjects::Base
 
         element :locations, "[data-qa=provider__locations]", text: "Locations"
         element :courses, "[data-qa=provider__courses]", text: "Courses"
-        element :current_cycle, "[data-qa=provider__courses__current_cycle]", text: "Current cycle (2019 - 2020)"
-        element :next_cycle, "[data-qa=provider__courses__next_cycle]", text: "Next cycle (2020 - 2021)"
+        element :current_cycle, "[data-qa=provider__courses__current_cycle]", text: "New cycle (2020 – 2021)"
```
Comment (defong) [inline spec/site_prism/page_objects/page/organisations/organisation_page.rb]: will `New` be filpped back to `Current`?

Comment (dankmitchell) [reply spec/site_prism/page_objects/page/organisations/organisation_page.rb]: Probably from Oct 1st. This design needs more work

File: spec/requests/sites_spec.rb (hunk 10 above)
Comment (danielburnley) [reply spec/requests/sites_spec.rb]: Not sure variables can be used in `it` names :thinking: I'd go more generic with "Redirects to /locations for the current recruitment cycle", then it doesn't need to be updated for every year
